intersect.py

```diff
@@ -14,16 +18,38 @@
 LAT_REGEX = re.compile('\\blat(itude)?\\b', flags=re.I)
 LNG_REGEX = re.compile('\\blo?ng(itude)?\\b', flags=re.I)
 
+# handle geo/json shape files
+
+def explode_shape_data(path):
+    """Break down multipolygons in geojson to single polygons per row.
+    Code from here: https://gist.github.com/mhweber/cf36bb4e09df9deee5eb54dc6be74d26"""
+    indf = gpd.GeoDataFrame.from_file(path)
+    outdf = gpd.GeoDataFrame(columns=indf.columns)
+    for idx, row in indf.iterrows():
+        if type(row.geometry) == Polygon:
+            outdf = outdf.append(row,ignore_index=True)
+        if type(row.geometry) == MultiPolygon:
+            multdf = gpd.GeoDataFrame(columns=indf.columns)
+            recs = len(row.geometry)
+            multdf = multdf.append([row]*recs,ignore_index=True)
+            for geom in range(recs):
+                multdf.loc[geom,'geometry'] = row.geometry[geom]
+            outdf = outdf.append(multdf,ignore_index=True)
+    return outdf
 
 def shapes_df(path):
-    zones = gpd.read_file(path).to_crs(fiona.crs.from_epsg(EPSG))
-    return zones.to_crs(CRS)
+    """Convert raw geojson data to geopandas dataframe with correct coordinates per CRS/EPSG"""
+    raw_shapes = explode_shape_data(path)
+    raw_shapes.crs = fiona.crs.from_epsg(EPSG) # set the crs on the dataframe to prevent "naive geometry" error
+    return raw_shapes.to_crs(CRS)
 
 
+# process people csv data
+
 def find_key(keys, regex):
     results = filter(regex.match, keys)
-    return next(results, None)
-
+    results_iter = (result for result in results)
+    return next(results_iter, None)
```

Comment:
Nice workaround! Does this mean that `find_key()` is sometimes getting passed a list rather than an iterator, or is this the Python 2/3 thing mentioned in https://github.com/nycivicengagementtable/spatial_join/pull/9#issuecomment-289263898? If the former, you might add a comment saying "normalizing to be an iterator" or something.